enhancements/installer/openstack-ipi-failure-domains.md

```diff
@@ -0,0 +1,401 @@
+---
+title: openstack-ipi-failure-domains
+authors:
+  - @EmilienM
+  - @mandre
+  - @mdbooth
+reviewers:
+  - @sdodson
+  - @staebler
+  - @JoelSpeed
+approvers:
+  - @sdodson
+  - @staebler
+api-approvers:
+  - "None"
+creation-date: 2022-07-04
+last-updated: 2022-07-04
+tracking-link:
+  - "https://issues.redhat.com/browse/OSASINFRA-2908"
+see-also:
+  - "/enhancements/installer/aws-customer-provided-subnets.md"
+  - https://github.com/openshift/enhancements/pull/918
+superseded-by:
+  - "https://docs.google.com/document/d/1_AjhGEyK6vGBFSSqiLVtLt_qOq6o9t2dtKqq2eFr48Y/"
+---
+
+# Allow to stretch a cluster across multiple failure domains
+
+## Summary
+
+For organizations where service availability is a concern, networking architectures are usually more modern than what we can find in a traditional datacenter.
+For example, the "Spine and Leaf" design is recommended choice when deploying OpenStack at scale.
+In this architecture, each leaf (lower-level access switch) is connected to multiple spines (upper-level core switch) in a full mesh. The latency becomes predictable, there is no more bottleneck and overall the resilience is improved.
+Each leaf brings their own network fabrics that are usually using its own power unit. Also each network fabrics manage their own Layer-2 domains, and therefore network subnets. Routing protocols are usually used so applications can communicate each other in the datacenter, since they don't live within the same domain anymore.
+
+In this enhancement, we would like to support the deployment of OpenShift into multiple subnets from day 1, using IPI.
+To accomplish this, the OpenStack administrator will pre-create a Routed Provider Network and one or multiple subnets per leaf, which will be linked to an Availability Zone, here called Failure Domain for a better understanding.
+Each Failure Domain will have its own network subnet and storage resources used by OpenShift.
+This feature will remove the limitation that we have today in BYON (Bring your own network) which is the support of a single primary subnet for the machines on day 1. We want to be able to deploy the cluster across multiple subnets and storage sources, where each resource live in a given Failure Domain.
+
+## Motivation
+
+TODO
+
+### User Stories
+
+As an administrator, I would like to stretch my OpenShift cluster across multiple Failure Domains so I increase the cluster resilience and performance.
```

Comment:
Need to make it explicit that this includes the control plane, and that the control plane will also have independent network fabrics. Possibly multiple user stories.